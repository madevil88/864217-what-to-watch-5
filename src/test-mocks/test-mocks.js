const reviews = [{
  "id": 1,
  "user": {
    "id": 1,
    "name": `Emely`
  },
  "rating": 4,
  "comment": `This movie is just plain bad. There must be some big payola going round this awards season.`,
  "date": `2020-11-24T16:25:11.950Z`
}, {
  "id": 2,
  "user": {
    "id": 3,
    "name": `Max`
  },
  "rating": 3,
  "comment": `I personally found this movie to be boring.`,
  "date": `2020-11-27T21:02:38.606Z`
}];

const films = {
  allFilms: [{
    "name": `Bronson`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    "background_color": `#73B39A`,
    "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    "rating": 3.6,
    "scores_count": 109661,
    "director": `Nicolas Winding Refn`,
    "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    "run_time": 92,
    "genre": `Action`,
    "released": 2008,
    "id": 1,
    "is_favorite": false,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }, {
    "name": `Pulp Fiction`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    "background_color": `#795433`,
    "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    "rating": 1.5,
    "scores_count": 1635992,
    "director": `Quentin Tarantino`,
    "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    "run_time": 153,
    "genre": `Crime`,
    "released": 1994,
    "id": 2,
    "is_favorite": true,
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }],
  filteredFilms: [{
    "name": `Bronson`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg`,
    "background_color": `#73B39A`,
    "description": `A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.`,
    "rating": 3.6,
    "scores_count": 109661,
    "director": `Nicolas Winding Refn`,
    "starring": [`Tom Hardy`, `Kelly Adams`, `Luing Andrews`],
    "run_time": 92,
    "genre": `Action`,
    "released": 2008,
    "id": 1,
    "is_favorite": false,
    "video_link": `http://media.xiph.org/mango/tears_of_steel_1080p.webm`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }, {
    "name": `Pulp Fiction`,
    "poster_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg`,
    "preview_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg`,
    "background_image": `https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg`,
    "background_color": `#795433`,
    "description": `The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.`,
    "rating": 1.5,
    "scores_count": 1635992,
    "director": `Quentin Tarantino`,
    "starring": [`John Travolta`, `Uma Thurman`, `Samuel L. Jackson`],
    "run_time": 153,
    "genre": `Crime`,
    "released": 1994,
    "id": 2,
    "is_favorite": true,
    "video_link": `http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4`,
    "preview_video_link": `https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm`
  }]
};

export {
  reviews,
  films
};
