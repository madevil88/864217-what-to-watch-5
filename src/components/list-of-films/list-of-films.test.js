import React from "react";
import ShallowRenderer from "react-test-renderer/shallow";
import ListOfFilms from "./list-of-films";
import {films} from "../../test-mocks/test-mocks";

const renderer = new ShallowRenderer();

it(`Should ListOfFilms render correctly`, () => {
  renderer.render(
      <ListOfFilms
        filteredFilms={films.filteredFilms}
        filmsCount={8}
      />
  );

  const tree = renderer.getRenderOutput();
  expect(tree).toMatchSnapshot();
});
