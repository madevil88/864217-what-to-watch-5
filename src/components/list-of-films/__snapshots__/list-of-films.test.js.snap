// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should ListOfFilms render correctly 1`] = `
Array [
  <React.Fragment>
    <WithActiveItem
      film={
        Object {
          "background_color": "#73B39A",
          "background_image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/bronson.jpg",
          "description": "A young man who was sentenced to seven years in prison for robbing a post office ends up spending three decades in solitary confinement. During this time, his own personality is supplanted by his alter-ego, Charles Bronson.",
          "director": "Nicolas Winding Refn",
          "genre": "Action",
          "id": 1,
          "is_favorite": false,
          "name": "Bronson",
          "poster_image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/bronson.jpg",
          "preview_image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/bronson.jpg",
          "preview_video_link": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
          "rating": 3.6,
          "released": 2008,
          "run_time": 92,
          "scores_count": 109661,
          "starring": Array [
            "Tom Hardy",
            "Kelly Adams",
            "Luing Andrews",
          ],
          "video_link": "http://media.xiph.org/mango/tears_of_steel_1080p.webm",
        }
      }
      id={1}
      initialActiveItem={-1}
    />
  </React.Fragment>,
  <React.Fragment>
    <WithActiveItem
      film={
        Object {
          "background_color": "#795433",
          "background_image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/background/Pulp_Fiction.jpg",
          "description": "The lives of two mob hitmen, a boxer, a gangster & his wife, and a pair of diner bandits intertwine in four tales of violence and redemption.",
          "director": "Quentin Tarantino",
          "genre": "Crime",
          "id": 2,
          "is_favorite": true,
          "name": "Pulp Fiction",
          "poster_image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/poster/Pulp_Fiction.jpg",
          "preview_image": "https://assets.htmlacademy.ru/intensives/javascript-3/film/preview/pulp-fiction.jpg",
          "preview_video_link": "https://upload.wikimedia.org/wikipedia/commons/transcoded/b/b3/Big_Buck_Bunny_Trailer_400p.ogv/Big_Buck_Bunny_Trailer_400p.ogv.360p.webm",
          "rating": 1.5,
          "released": 1994,
          "run_time": 153,
          "scores_count": 1635992,
          "starring": Array [
            "John Travolta",
            "Uma Thurman",
            "Samuel L. Jackson",
          ],
          "video_link": "http://peach.themazzone.com/durian/movies/sintel-1024-surround.mp4",
        }
      }
      id={2}
      initialActiveItem={-1}
    />
  </React.Fragment>,
]
`;
